hybrasyl/Messaging/Group.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Hybrasyl.Objects;
+
+
+namespace Hybrasyl.Messaging
+{
+
+    class GroupCommand : ChatCommand
+    {
+        public new static string Command = "group";
+        public new static string ArgumentText = "<string username>";
+        public new static string HelpText = "Invite the specified player to your group.";
+        public new static bool Privileged = false;
+
+        public new ChatCommandResult Run(User user, params string[] args)
```

Comment:
Missing static declaration here. (other's require as well)

Reply:
`ChatCommand` should become `IChatCommand`, and from there each individual "command" class should implement `IChatCommand`. These methods do not need to be static.

On a side note, `public new` is incorrect syntax. `public static new` would be correct.

Reply:
example:
`public interface IChatCommand
    {
        string Command { get; }
        string ArgumentText { get; }
        string HelpText { get; }
        bool Privileged { get; }
        int ArgumentCount { get; }
        ChatCommandResult Success(string ErrorMessage = null, byte MessageType = MessageTypes.SYSTEM);
        ChatCommandResult Fail(string ErrorMessage, byte MessageType = MessageTypes.SYSTEM);
        ChatCommandResult Run(User user, params string[] args);
    }`

Reply:
The decision to implement chat commands in this fashion was a conscious one. 

There is no need for these methods to have instances; they do not act on themselves and only have static variables.

Reply:
agreed, but the implemented abstract class can become an interface, while the resulting inheriting class can become static, implementing the members of the interface.

Reply:
You cannot use interfaces with static classes.

Reply:
correct, however you can use a static interface.

Reply:
At least in C# 7, `public static interface` or thereabouts is illegal.